_submitplugins/rrSubmit_Nuke_5.py

```diff
@@ -162,16 +161,20 @@ def writeToXMLEnd(self, f,rootElement):
         xml = ElementTree(rootElement)
         self.indent(xml.getroot())
 
-        if not f == None:
-            xml.write(f)
-            f.close()
-        else:
-            print("No valid file has been passed to the function")
+        if f is None:
+            print("No valid file has been passed to the write function")
             try:
                 f.close()
             except:
                 pass
             return False
+
+        if sys.version_info.major == 2:
+            xml.write(f)
+        else:
+            xml.write(f, encoding='unicode')
```

Comment:
I don't know if this is necessary now that you have changed to using tempfile in binary mode.